### PR TITLE
Alpha svein

### PR DESCRIPTION
**Pull request by kongebra**
# Refactored
* Form Builder for reviews and submissions

# Added
* Ability to delete reviews and submissions

# Note
```
$ golint ./...
shared\session\session.go:103:1: exported function SaveMessageToSession should have comment or be unexported
shared\session\session.go:125:1: exported function GetAndDeleteMessageFromSession should have comment or be unexported
shared\session\session.go:148:1: exported function DeleteMessageFromSession should have comment or be unexported
```
These are not mine

### CODE REVIEW

**Comment by kongebra** [comment]
> * If I change Review Form from ex ? `WhoRuForm` to none, it wont change
> * When trying to delete or edit a submission/review form, it should ask if the user is sure and show how many has answered the forms
> * When I'm going to review an assignment, it only shows me the submitted assignment and not a review form, so I can't check if it's working or not
> * Not sure if you're have implemented it, but if admin wants to change review form in assignment edit, it should act the same way as with submission form (If there are more than 0 students that has answered the form, give warning about how many and if admin still wants to switch, delete former reviews)

* Fixed
* Not using more time on this then needed, `alpha delivery` > `new features`
* Fixed
* Not using more time on this then needed, `alpha delivery` > `new features`

File: webservice/shared/view/plugin/deadlinedue_test.go [hunk 1]
```diff
@@ -10,40 +10,55 @@ import (
 )
 
 func TestDeadlineDue(t *testing.T) {
+<<<<<<< HEAD
```
Comment (BredeFK) [inline webservice/shared/view/plugin/deadlinedue_test.go]: fix merge error

File: webservice/shared/view/plugin/deadlinedue_test.go [hunk 2]
```diff
@@ -10,40 +10,55 @@ import (
...
 
 	result := buffer.String()
 
-	if result != "true" {
+	if result != expected {
+		t.Logf("expected: %v, got: %v", expected, result)
 		t.Fail()
 	}
 
+<<<<<<< HEAD
```
Comment (BredeFK) [inline webservice/shared/view/plugin/deadlinedue_test.go]: fix merge error

File: webservice/shared/view/plugin/prettytime_test.go [hunk 3]
```diff
@@ -12,31 +12,36 @@ import (
...
 
-	err = tmpl.Execute(buffer, data)
+	err = tmpl.Execute(buffer, input)
 	if err != nil {
 		log.Fatalf("execution: %s", err)
 	}
 
 	result := buffer.String()
 
+<<<<<<< HEAD
```
Comment (BredeFK) [inline webservice/shared/view/plugin/prettytime_test.go]: fix merge error

**Pull request merged**